core/node/node_framework/src/implementations/resources/pools.rs

```diff
@@ -95,10 +95,14 @@ impl ReplicaPoolResource {
     }
 
     pub async fn get_singleton(&self) -> anyhow::Result<ConnectionPool> {
+        self.get_custom(1).await
+    }
+
+    pub async fn get_custom(&self, size: u32) -> anyhow::Result<ConnectionPool> {
         let result = self.builder.build_singleton().await;
 
         if result.is_ok() {
-            self.connections_count.fetch_add(1, Ordering::Relaxed);
+            self.connections_count.fetch_add(size, Ordering::Relaxed);
             let total_connections = self.connections_count.load(Ordering::Relaxed);
```

Comment:
Nit: Since `fetch_add` returns the previous value, it could make sense to compute total connections as `fetch_add(_) + size` for atomicity. Not a big deal, obviously.